Added a page with a short tutorial for the projectSdkSetupValidator EP.

takanuva15 opened the pull request:
Added a quick code tutorial in a sub-page to the "Assisting in Setting Up an SDK" section to help new plugin developers (like me) to quickly get running with this great EP. I formatted the files/links according to the code style guidelines so hopefully everything should be formatted correctly.

halirutan commented on topics/reference_guide/sdk_setup_assistance.md:
```diff
@@ -0,0 +1,35 @@
+
+<!-- Copyright 2000-2020 JetBrains s.r.o. and other contributors. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file. -->
+
+Prompting the user with a notification to set up an SDK can help them get up-and-running with a plugin faster. The IntelliJ Platform offers extension point `com.intellij.projectSdkSetupValidator`, where you can register an implementation of [`ProjectSdkSetupValidator`](upsource:///platform/lang-impl/src/com/intellij/codeInsight/daemon/ProjectSdkSetupValidator.java) to notify the user if they are  ...
```
I'd add an article here: "The IntelliJ Platform offers **the** extension point ..."

Unless, of course, you are a native speaker and you know better :)

takanuva15 replied:
Lol I am a native speaker, but I still racked my brain for 10 minutes thinking whether the original sentence was incorrect. I think putting "the" is more correct though, since a similar sentence like "The restaurant offers {} menu option 'escargot'..." wouldn't make sense without "the" at {}. (English is the weirdest language).

I'll push up a change now to fix that.

halirutan commented:
Looks pretty nice to me. Two points:

1. I don't think we need a new page for this. It's not too long and can still fit at the bottom of the existing "SDK" page.
2. We try to prevent using comments in code-blocks. What I propose instead are the following two steps:
    1. Turn your comments into a list above or below the code block where you basically say the same, just a little more verbose. Like "`methodX()` will check if the condition YY is met...". You find an example of how it's done in other places [like right above the header here.](https://plugins.jetbrains.com/docs/intellij/element-patterns.html#tools-and-debugging)
    2. We fix the JavaDocs in the IntellJ sources for this interface along the way. It would be awesome if you could turn your comments on the methods into suggestions for JavaDoc [on the interface](https://upsource.jetbrains.com/idea-ce/file/idea-ce-ba0c8fc9ab9bf23a71a6a963cd84fc89b09b9fc8/platform/lang-impl/src/com/intellij/codeInsight/daemon/ProjectSdkSetupValidator.java?_ga=2.98877655.892746014.1615735196-101771966.1615280212). Also, since you're at it, I'm pretty sure you have an idea for the JavaDoc comment of the interface itself. It doesn't need to be anything fancy. Just say a few words and Yann and myself go over it and put in our two cents. What we usually do is to link from the JavaDocs to the SDK docs in the following way:
       ```
       Please see the <a href="https://plugins.jetbrains.com/docs/intellij/element-patterns.html">IntelliJ Platform Docs</a>
       for a high-level overview.
       ```

If you're up to put in some more work, I'm sure we'll have this published by the end of the week. Thanks for your effort!

takanuva15 commented:
@halirutan 

> Looks pretty nice to me. Two points:
> 
> 1. I don't think we need a new page for this. It's not too long and can still fit at the bottom of the existing "SDK" page.

I've amended the commit and moved all my changes back into the original `sdk.md` file

> 2. We try to prevent using comments in code-blocks. What I propose instead are the following two steps:
>    
>    1. Turn your comments into a list above or below the code block where you basically say the same, just a little more verbose. Like "`methodX()` will check if the condition YY is met...". You find an example of how it's done in other places [like right above the header here.](https://plugins.jetbrains.com/docs/intellij/element-patterns.html#tools-and-debugging)

Done. I've deleted the original doc comments and added a list below the code explaining what's happening.

>    2. We fix the JavaDocs in the IntellJ sources for this interface along the way. It would be awesome if you could turn your comments on the methods into suggestions for JavaDoc [on the interface](https://upsource.jetbrains.com/idea-ce/file/idea-ce-ba0c8fc9ab9bf23a71a6a963cd84fc89b09b9fc8/platform/lang-impl/src/com/intellij/codeInsight/daemon/ProjectSdkSetupValidator.java?_ga=2.98877655.892746014.1615735196-101771966.1615280212).

I'm not sure I understand. Are you saying I should raise a PR to the official `intellij-community` repo on GitHub with doc comments added to that interface & its methods? Or is there some other way I should be proposing those comments?

halirutan commented:
> I'm not sure I understand. Are you saying I should raise a PR to the official intellij-community repo on GitHub with doc comments added to that interface & its methods? Or is there some other way I should be proposing those comments?

We would take this off your shoulders and Yann or I would take care of committing your JavaDocs to the official repo. There are several ways you can share your ideas. The easiest one is the following:

- You simply open up your local IntelliJ sources, edit the `ProjectSdkSetupValidator` file and add your JavaDoc
- Create a patch file and put it online as a Gist (not sure you can attach files here) and share the link.

For the unlikely situation that you don't know how to create a patch file from your local chances, just comment here and I'll give you a quick walk-through.

YannCebron commented:
Thanks for your contribution, LGTM. added 2 minor comments

takanuva15 commented:
@YannCebron 
> one empty line before code section is required

Done; amended the commit. (The Jetbrains team should add a linter for this so you won't even need to tell people to do this. I'd imagine a bash script to check this among all `*.md` files in the repo shouldn't be too hard to implement)

> replace "will ..." with present tense

Done; amended the commit

@halirutan I have created a gist with the new javadoc proposal for `ProjectSdkSetupValidator` as you have mentioned. See here:  https://gist.github.com/takanuva15/3e51f5fd06fe2c6c60681e57ac1ca810

YannCebron commented:
@takanuva15 please sign CLA so we can use your javadoc gist for IJ Community sources https://www.jetbrains.com/agreements/cla/ Thanks!

halirutan commented:
@takanuva15 Thanks again for your PR. I'll have a look at your JavaDoc suggestions in the upcoming days and will merge it to the repo.

One more thing: Are you a member of [our IntelliJ Platform Slack channel](https://plugins.jetbrains.com/slack)? If yes, please write me a direct message.
